horizomer/utils/tree.py

```diff
@@ -264,6 +279,30 @@ def root_above(node, name=None):
     return res
 
 
+def _exact_compare(tree1, tree2):
+    """Simultaneously compares the name, length, and support of each node from
+    two trees.
+
+    Parameters
+    ----------
+    tree1: skbio.TreeNode
+        first tree to compare
+    tree2: skbio.TreeNode
+        second tree to compare
+
+    Returns
+    -------
+    bool
+        `True` if name, length, and support of each corresponding node are same
+        `False` otherwise
+    """
+    for n1, n2 in zip(tree1.postorder(), tree2.postorder()):
+        if (n1.name != n2.name or n1.length != n2.length or
```

Comment:
One issue: A TreeNode object by default does not have the `support` attribute. So we need to use `hasattr` to test this before comparing.

Reply:
Or maybe use [`getattr`](https://docs.python.org/3/library/functions.html#getattr) if you want to make things simple.